spec/icmp_spec.cr

```diff
@@ -36,6 +37,52 @@ describe "icmp" do
     yield_count.should eq ping_count
   end
 
+  it "honours short inter-ping delay" do
+    ping_count = 3
+    delay = 0.1
+
+    start_time = Time.now
+    
+    ICMP::Ping.new(HOST).ping(count: ping_count, delay: delay) do |request|
+    end
+
+    elapsed = (Time.now - start_time).to_f
+
+    elapsed.should be >= delay * (ping_count-1)
```

Comment:
Comparing floats like this in specs feels risky to me. I worry it will lead to a flaky test. I think your math is right, but the test won't fail if the delay is implemented wrong in the tested code. 

Crystal Spec::Expectations has a [be_close](https://crystal-lang.org/api/0.24.2/Spec/Expectations.html#be_close%28expected%2Cdelta%29-instance-method) method that might be flexible enough to be used here.

Reply:
Thanks - I didn't know about #be_close (or &measure above). I'll re-implement with those and commit changes.

My software dev inexperience is showing...